### PR TITLE
WIP kafka api changes

### PR DESCRIPTION
**Pull request by VladoLavor**
Signed-off-by: Vladimir Lavor <vlavor@cisco.com>

### CODE REVIEW

File: examples/kafka_plugin/main.go [hunk 1]
```diff
@@ -135,6 +137,14 @@ func (plugin *ExamplePlugin) Init() (err error) {
 		plugin.Log.Error(err)
 	}
 
+	// todo add to example
+	//var partition int32 = 0
+	//manualSyncPublisher := connection.NewManualSyncPublisher(topic, partition)
```
Comment (jozef-slezak) [inline examples/kafka_plugin/main.go]: Let's be consistent with naming ConsumeParition(). What about
NewSyncPublisherToPartition() instead NewManualSyncPublisher()?

File: messaging/kafka/mux/proto_connection.go [hunk 2]
```diff
@@ -110,6 +110,14 @@ func (conn *ProtoConnection) ConsumeTopic(msgClb func(messaging.ProtoMessage), t
 	return nil
 }
 
+// ConsumePartition is called to start consuming given topic on given partition and offset.
+// Function can be called until the multiplexer is started, it returns an error otherwise.
+// The provided channel should be buffered, otherwise messages might be lost.
+func (conn *ProtoConnection) ConsumePartition(msgClb func(messaging.ProtoMessage), topic string, partition int32, offset int64) error {
```
Comment (jozef-slezak) [inline messaging/kafka/mux/proto_connection.go]: let's copy&paste constants for latest and oldest offset

**Comment by lukasmacko** [comment]
please remove binary `examples/kafka_plugin/clustered/clustered`

File: examples/kafka_plugin/clustered/main.go [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+	ef.Inject()
+	return core.ListPluginsInFlavor(ef)
+}
+
+/******************
+ * Example plugin *
+ ******************/
+
+// ExamplePlugin implements Plugin interface which is used to pass custom plugin instances to the agent. The Kafka
+// ConsumerHandle is required to read messages from a topic, and PluginConnection is needed to start consuming on
```
Comment (lukasmacko) [inline examples/kafka_plugin/clustered/main.go]: This comment seems to be outdated.

**Pull request merged**